crates/matrix-sdk/src/send_queue.rs

```diff
@@ -673,6 +693,19 @@ impl QueueStorage {
             .await?)
     }
 
+    /// Marks an event identified with the given transaction id as being now
+    /// unwedged and adds it back to the queue.
+    async fn mark_as_unwedged(
+        &self,
+        transaction_id: &TransactionId,
+    ) -> Result<(), RoomSendQueueStorageError> {
+        Ok(self
+            .client()?
+            .store()
+            .update_send_queue_event_status(&self.room_id, transaction_id, false)
```

Comment:
Hmm I'm wondering if this function should return a boolean, to indicate if the underlying item with such a transaction id was found, otherwise this might give a false sense that something happened, while it's not the case… (if one passes an invalid transaction id, for instance)

We don't have to do it in this PR, though.

Reply:
Oh yeah, that would make sense but why not return an error directly?

Reply:
It's a philosophical question: is it an error to misuse the API with a transaction id that's not handled by the send queue, or should we signal it nicely so that the user can try something else with that transaction id?

Reply:
I would passing in an invalid/unknown transaction id should be a developer error, api missuse. Giving back a boolean might be unclear to what actually went wrong.